File: DotNet/Framework/AutomationFramework/Pages/LoginPage.cs
using Helpers.Actions;
using OpenQA.Selenium;

namespace Pages
{
    public sealed class LoginPage
    {
        private static Lazy<LoginPage> GetLoginPage = new Lazy<LoginPage>(() => new LoginPage());
        private static BasePage Base_Page;
        private static DriverActions driverActions;
        public static LoginPage Login_Page { get 
            {
                Base_Page = BasePage.Base_Page;
                driverActions = Base_Page.driverActions;
                return GetLoginPage.Value;  
            } 
        }

        private LoginPage() 
        {
        }

        #region Locators

        private By Logo = By.CssSelector(".orangehrm-login-branding img");
        private By UserName = By.Name("username");
        private By Password = By.Name("password");
        private By Login = By.XPath("//button[@type='submit']");

        #endregion

        
        #region Actions

        public LoginPage IsLogoPresent(out bool isPresent)
        {
            isPresent = driverActions.GetPresenceOfElement(Logo);
            return this;
        }

        public LoginPage WaitForPageLoad()
        {
            driverActions.WaitUntilElementClickable(Logo);
            return this;
        }

        public LoginPage EnterUsername(string username)
        {
            driverActions.WaitUntilElementDisplayed(UserName);
            driverActions.keyBoardActions.SendText(UserName, username);
            return this;
        }

        public LoginPage EnterPassword(string password)
        {
            driverActions.WaitUntilElementDisplayed(Password);
            driverActions.keyBoardActions.SendText(Password, password);
            return this;
        }

        public DashboardPage ClickLogin()
        {
            driverActions.WaitUntilElementClickable(Login);
            driverActions.mouseActions.ClickElement(Login);
            return DashboardPage.Dashboard_Page;
        }

        #endregion
    }
}

File: Java/AutomationFramework/src/com/app/pages/LoginPage.java
package com.app.pages;

import com.app.helpers.actions.DriverActions;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage {
    private static BasePage Base_Page;
    private static LoginPage Login_Page;
    public static DriverActions driverActions;
    public static WebDriver driver;

    private LoginPage() {
    }

    public static LoginPage GetLoginPage() {
        if(Login_Page == null) {
            synchronized (LoginPage.class) {
                if(Login_Page == null)
                    Login_Page = new LoginPage();
            }
        }
        Base_Page = BasePage.GetBasePage();
        driverActions = Base_Page.driverActions;
        return Login_Page;
    }

    public LoginPage Init_Elements() {
        PageFactory.initElements(Base_Page.driver, this);
        return this;
    }

    @FindBy(css = ".orangehrm-login-branding img")
    private WebElement Logo;
    @FindBy(name = "username")
    private WebElement UserName;
    @FindBy(name = "password")
    private WebElement Password;
    @FindBy(xpath = "//button[@type='submit']")
    private WebElement Login;

    public boolean IsLogoPresent()
    {
        boolean isPresent = driverActions.GetPresenceOfElement(Logo);
        return isPresent;
    }

    public LoginPage WaitForPageLoad()
    {
        driverActions.WaitUntilElementVisibleAndClickable(Logo);
        return this;
    }

    public LoginPage EnterUsername(String username)
    {
        driverActions.WaitUntilElementVisibleAndClickable(UserName);
        driverActions.keyBoardActions.SendText(UserName, username);
        return this;
    }

    public LoginPage EnterPassword(String password)
    {
        driverActions.WaitUntilElementVisibleAndClickable(Password);
        driverActions.keyBoardActions.SendText(Password, password);
        return this;
    }

    public DashboardPage ClickLogin()
    {
        driverActions.WaitUntilElementVisibleAndClickable(Login);
        driverActions.mouseActions.ClickElement(Login);
        return DashboardPage.GetDashboardPage()
                .Init_Elements();
    }
}
